src/sys/select.rs

```diff
@@ -1,70 +1,38 @@
-use std::ptr::null_mut;
+use std::mem;
 use std::os::unix::io::RawFd;
-use libc::{c_int, timeval};
+use std::ptr::null_mut;
+use libc::{self, c_int};
 use {Errno, Result};
 use sys::time::TimeVal;
 
-pub const FD_SETSIZE: RawFd = 1024;
-
-#[cfg(any(target_os = "macos", target_os = "ios"))]
-#[repr(C)]
-#[derive(Clone)]
-pub struct FdSet {
-    bits: [i32; FD_SETSIZE as usize / 32]
-}
-
-#[cfg(any(target_os = "macos", target_os = "ios"))]
-const BITS: usize = 32;
+pub use libc::FD_SETSIZE;
 
-#[cfg(not(any(target_os = "macos", target_os = "ios")))]
+// FIXME: Change to repr(transparent) once it's stable
 #[repr(C)]
-#[derive(Clone)]
-pub struct FdSet {
-    bits: [u64; FD_SETSIZE as usize / 64]
-}
-
-#[cfg(not(any(target_os = "macos", target_os = "ios")))]
-const BITS: usize = 64;
+pub struct FdSet(libc::fd_set);
 
 impl FdSet {
     pub fn new() -> FdSet {
-        FdSet {
-            bits: [0; FD_SETSIZE as usize / BITS]
-        }
+        let mut fdset = unsafe { mem::uninitialized() };
+        unsafe { libc::FD_ZERO(&mut fdset) };
```

Comment:
This shouldn't require `unsafe`.  `FD_ZERO` is implemented in pure Rust.  Same for `FD_CLR`, `FD_SET`, and `FD_ISSET`